### PR TITLE
[Serve] Rename `input_schema` to `http_adapter` and clarify it in doc

### LINKED ISSUE #24192: [Serve] Rename `input_schema` to `http_adapter`
We are using `input_schema` as keyword argument in both DAGDriver and ModelWrapper. This field is confusing for users. We should rename them to http_adapter for UX benefit.

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR should be a simple rename. But I also went into to clarify the HTTP adapter documentation further. 

## Related issue number
Closes #24192

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_model_wrappers.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def test_yaml_compatibility(serve_instance):
                             "checkpoint_cls": checkpoint_cls,
                             "uri": path,
                         },
-                        "input_schema": schema_func,
+                        "http_adapter": schema_func,
```
Comment (jiaodong) [inline python/ray/serve/tests/test_model_wrappers.py:145]: schema_func -> adapter_func for this and line 130 ?

File: doc/source/serve/http-servehandle.rst [hunk 2]
```diff
@@ -160,13 +160,36 @@ HTTP Adapters
...
+
+    async def json_resolver(request: starlette.requests.Request):
+        return await request.json()
+
+Here is an adapter that accept two HTTP query parameters.
+
+.. code-block:: python
+
+    def parse_query_args(field_a: int, field_b: str):
+        return YourDataClass(field_a, field_b)
```
Comment (jiaodong) [inline doc/source/serve/http-servehandle.rst:178]: can we put up a full, short snippet here to show user what the http request look like and how its resolved by http_adapter function into `YourDataClass` ?

**Review by jiaodong — APPROVED** [review]
looks good to me, just some nits. Will leave it to you to decide when to merge.

Btw are you trying to include this in 1.13 branch cut ?

**Comment by simon-mo** [comment]
> Btw are you trying to include this in 1.13 branch cut ?
yes

**Pull request merged**